core/types/transaction.go

```diff
@@ -96,6 +93,9 @@ type TxData interface {
 	// copy of the computed value, i.e. callers are allowed to mutate the result.
 	// Method implementations can use 'dst' to store the result.
 	effectiveGasPrice(dst *big.Int, baseFee *big.Int) *big.Int
+
+	encode(*bytes.Buffer) error
+	decode([]byte) error
```

Comment:
is it okay to not signal these are RLP encode / decode functions?

Reply:
They're not specifically for RLP, these could also be implemented with another serialization method.

Reply:
ah i see